### PR TITLE
Support registration

### PR DESCRIPTION
**Pull request by jicc96**
This PR adds the following endpoints:
'/api/v1/invite_support' : permits an administrator send a invitation via email.
'/api/v1/confitmation' : permits the completion of the support registration. 
'/api/v1/confitmation/verify' : permits the verification of the confirmation token validity. 
Associated cards: https://trello.com/c/qdam7QKf | https://trello.com/c/ES1mWOpH | https://trello.com/c/6F6EXrM9 .
All test pass with a coverage of 98.72% and no rubocop warnings.

### CODE REVIEW

File: app/controllers/api/v1/confirmations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        if password.present? && password_confirmation.present? && password == password_confirmation
+          support = User.find_by(confirmation_token: confirmation_token)
+          return render json: { error: 'The provided token is invalid.' }, status: :bad_request unless support.present?
+          return render json: { error: 'The user is alredy confirmed, try logging in.' }, status: :bad_request if support.confirmed?
+          return render json: { error: 'That token has expired.' }, status: :bad_request if support.period_expired?
+
+          support.first_name = first_name
+          support.last_name =  last_name
+          support.password = password
+          return render json: { error: 'Could not modify the user, you must try again later.', error_messages: support.errors.messages }, status: :internal_server_error unless support.save
```
Comment (petterboussard) [inline app/controllers/api/v1/confirmations_controller.rb]: Are we sure that we want to send a 500?

File: app/controllers/api/v1/confirmations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+          return render json: { error: 'Could not confirm the user, you must try again later.', error_messages: support.errors.messages }, status: :internal_server_error unless support.confirm && support.save
```
Comment (petterboussard) [inline app/controllers/api/v1/confirmations_controller.rb]: Same here.

File: app/controllers/api/v1/user_controller.rb [hunk 3]
```diff
@@ -20,6 +20,39 @@ def create
...
+                render json: { error: 'The user already accepted' }, status: :gone
+              else
+                user.resend_confirmation_instructions
+                render json: { message: 'Re-sending invitation to user.' }, status: :ok
+              end
+            else
+              render json: { error: 'The email provided is used.' }, status: :bad_request
+            end
+          else
+            invited = User.create(email: email, role: 'support', site_id: admin.site.id)
```
Comment (petterboussard) [inline app/controllers/api/v1/user_controller.rb]: Change this line to: 
invited = User.new(email: email, role: 'support', site_id: admin.site.id)

File: app/views/devise/mailer/confirmation_instructions.html.erb [hunk 4]
```diff
@@ -0,0 +1,2661 @@
...
+
+    
+
+
+  </head>
+  <body topmargin="0" leftmargin="0" marginheight="0" marginwidth="0" style="-ms-text-size-adjust: 100%; -webkit-text-size-adjust: 100%; background: #fafafa; color: #222222; font-family: 'Helvetica', 'Arial', sans-serif; font-size: 14px; font-weight: normal; line-height: 19px; margin: 0; min-width: 100%; padding: 0; text-align: left; width: 100% !important" bgcolor="#fafafa">
+<style type="text/css">
+body {
+width: 100% !important; min-width: 100%; -webkit-text-size-adjust: 100%; -ms-text-size-adjust: 100%; margin: 0; padding: 0;
+}
```
Comment (petterboussard) [inline app/views/devise/mailer/confirmation_instructions.html.erb]: Usually CSS and HTML are created as two different files. 
This change doesnt need to be done now, with more time we should totally do it.

File: config/environments/development.rb [hunk 5]
```diff
@@ -50,5 +50,13 @@
   # routes, locales, etc. This feature depends on the listen gem.
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
 
+  config.action_mailer.smtp_settings = {
+    :address => "smtp.gmail.com",
+    :port => 587,
+    :user_name => "no.replay.neocoast.backend@gmail.com",
+    :password => "lfkjsdhfliuwheliu947y58743.,we/r.tet/.rer.rt/er.",
```
Comment (petterboussard) [inline config/environments/development.rb]: Wow thats a strong password!

File: app/controllers/api/v1/user_controller.rb (hunk 3 above)
Comment (jicc96) [reply app/controllers/api/v1/user_controller.rb]: The change was implemented as requested

File: README.md [hunk 6]
```diff
@@ -163,6 +163,83 @@ If the creation is successfull the response indicates that the entities were cre
...
+  message: 'string'
+} 
+**Response Schema on error**: 
+{
+  error: 'string'
+}
+
+### Confirmation for support :  
+```
+post app_domain/api/v1/confitmation
```
Comment (guilleLara) [inline README.md]: It should be '/confirmation' not 'confitmation'

File: config/initializers/devise.rb [hunk 7]
```diff
@@ -139,7 +139,7 @@
   # their account can't be confirmed with the token any more.
   # Default is nil, meaning there is no restriction on how long a user can take
   # before confirming their account.
-  # config.confirm_within = 3.days
+  config.confirm_within = 1.hour
```
Comment (guilleLara) [inline config/initializers/devise.rb]: I think we should give a little more margin to confirm, for instance I don't usually see an email within an hour

**Pull request merged**